src/bindings/html/dom.js

```diff
@@ -48,7 +48,9 @@ export function translateMutations(view, langs, mutations) {
             if (addedNode.childElementCount) {
               getTranslatables(addedNode).forEach(targets.add.bind(targets));
             } else {
-              targets.add(addedNode);
+              if (addedNode.hasAttribute('data-l10n-id')) {
```

Comment:
Inserting an empty DocumentFragment does not trigger mutation observer, and we guard against non-element nodes.